### ISSUE TITLE
Text entered in "addNewLabel" fields is ignored by Theming, and Language Exceptions cannot be used to achieve the same goal

### ISSUE THREAD

**Issue by Rainribbon** [Kaltura-EDU/issues-tracking#19]
Submitted to Kaltura as Case #00549172:

When Theming is enabled, the text entered in the "addNewLabel" field in the following KAF modules is ignored:

* Recorder
* Kalturacapture

The default labels, "Express Capture" and "Capture," respectively, are meaningless and confusing to users. Other Kaltura admins with whom I have spoken have also used the "addNewLabel" fields to correct the confusing default labels in their instances of Kaltura.

Attempting to use a language exception to correct the labels is not a viable workaround because it impacts other areas of the interface in inappropriate ways. For example, if I try to replace "Capture" with "Record with Kaltura Capture" (the desired label), the text is cut off in the menu, as shown in the screenshot below.

![Image](https://github.com/user-attachments/assets/5d1160e2-03c1-429c-8316-fa1d42efc07c)

Additionally, other buttons and fields in the interface where the text "Capture" is used are overwritten and will provide wrong information as a result. For example, there is a "Capture" button used to replace a file's default thumbnail, which is inappropriately replaced via the language exception, as shown in the screenshot below.

![Image](https://github.com/user-attachments/assets/0a0fab7e-126e-4142-bfd8-f09ad18662a5)

Steps to reproduce:
1. In KAF admin, enable Theming.
2. Navigate to the "Recorder" module and enter text in the "addNewLabel" field to replace "Express Capture" and save.
3. Navigate to the "Kalturacapture" module and enter text in the "AddNewLabel" field to replace "Capture" and save.
4. In the LMS, access My Media, open the "Create" menu, and note that the changes have not taken effect.
5. Access Media Gallery in a course, open the "Create" menu, and note that the changes have not taken effect.

**Comment by Rainribbon** [comment]
I received this follow-up from Kaltura today:

> Thanks again for bringing this up. From our end, this is working as expected, fitting a longer label through Language Exceptions would fall under a feature request, as it’s more of a limitation than a bug. It may not be something worth investing too much energy into unless it’s a high priority on your side.
> 
> For now, it might be best to use a shorter label, like “Record” or “Record with Capture,” which fits better within the space constraints and avoids unintended impacts elsewhere in the interface.
> 
> If you’d like to move forward with a feature request, I’m happy to loop in your CSM. Otherwise, if you have any other questions, feel free to reach out, I’m happy to help.

My response:

> This isn't a request to make the menu width adjust to Language Exceptions. As I indicated, the Language Exceptions *cannot* be used as a workaround here. Using a Language Exception to change "Capture" changes that word in all contexts throughout the entire user interface, including the "Capture" button for capturing an icon which *must* be "Capture" and nothing else.
> 
> I consider it a feature parity issue and a bug that:
> 1. Theming does not recognize the text entered in the "addNewLabel" fields which have been provided to change the labels of the affected menu options. (We and other schools have been using these fields for years to address the confusing naming conventions that Kaltura uses.)
> 2. I assume that if the menu *did* properly utilize the "addNewLabel" text, it would not be responsive, which it needs to be.
> 
> Please see the attached screenshot for the menu options we have in our system now, which we would like to retain.
> 
> ![Image](https://github.com/user-attachments/assets/484322cb-eb4b-40bd-826a-ad0f7ffc606a)
> 
> That said, if Kaltura used meaningful, easy-to-understand menu and button labels throughout the system, rather than overly technical terminology in some places and vague and confusing terminology elsewhere to describe key features of the tool, institutions might not need options to change them.

**Comment by Rainribbon** [comment]
Our Kaltura engineer is looking into getting clarification on the expected behavior and interaction of the addNewLabel option versus Language Exceptions when theming is enabled.

I thanked them for investigating and added the following:

> If the decision is that Language Exceptions are the intended method of updating these labels, the labels must have unique message IDs. For example, "Capture" is currently the ID for both the menu item that opens the Kaltura Capture software and the label for the button to capture a thumbnail image, which results in some of the bad behavior I mentioned here.

**Comment by Rainribbon** [comment]
My ticket was updated indicating that the addNewLabel feature is currently only supported for non-theming use cases. However, the Product team will investigate whether support for theming is on the roadmap.

**Comment by alexumn** [comment]
At the University of Minnesota we submitted case# 00552530, "Theming/DS REACH UI Create menu language exception conflicts" on June 18, 2025. Here's what we said:

> We are testing the Theming/Design System UI. I believe portions of this bug are recorded in the University of Virginia's case# 00549172.
> 
> To align with our current Add New menu contents (pre-Theming implementation), we want to use language exceptions for "Capture" and "Upload". We cannot do that without also impacting other (non-Theming) areas in the user interface.
> 
> First and foremost, we need to change "Capture" to "Kaltura Capture app" because "Capture" is not clear to users. The associated English default.po contents are below:
> 
> #: modules/kalturacapture/models/Kalturacapture.php:197
> #: modules/thumbnails/views/scripts/index/index.ajax.phtml:41
> msgid "Capture"
> msgstr "Capture"
> 
> When we do that, the language exception also appears in the Edit page's Thumbnails tab, as seen in the attached screenshot "KMS Theming "Capture" exception appears in Thumbnails tab 20250618.png".
> 
> 
> The second is that we need to change "Upload" to "Upload Media". The associated English default.po contents are below:
> 
> #: application/views/scripts/partials/layout/headScript.phtml:445
> #: application/views/scripts/partials/usermedia/translations.phtml:30
> #: application/views/scripts/schedule-resource-admin/upload.dialog.phtml:32
> #: library/Kms/Helper/Upload/AddNewActions.php:94
> #: modules/chapters/views/scripts/partials/details/chapter.phtml:14
> #: modules/disclaimer/views/scripts/index/uploadmedia-module.phtml:3
> #: modules/uploadmedia/models/Uploadmedia.php:50
> #: modules/uploadmedia/views/scripts/index/index.phtml:2
> msgid "Upload"
> msgstr "Upload"
> 
> When we use a language exception to replace "Upload" with "Upload Media", that change also appears in the Edit page's Timeline tab when "Create a new chapter" is selected. Refer to the attached screenshot "KMS Theming "Upload" exception appears in Timeline tab 20250618.png".

We have not received a meaningful reply yet.

See below for the two screenshots.

![Image](https://github.com/user-attachments/assets/40543330-f850-4b92-ac00-bdbf32956e92)

![Image](https://github.com/user-attachments/assets/23450c2b-f7a4-4d11-b5da-97cfdcb432e3)

**Comment by Rainribbon** [comment]
I received this response today:

> I wanted to share an update from our Product team. At this time, there are no plans to support this functionality within theming. Based on their guidance, they recommend using language exceptions as a workaround in this case.

I sent the following reply:

> Thanks for following up! The problem is that currently, Language Exceptions are not supported as a workaround for this use case, either.
> 
> If Language Exceptions must be used, the text strings need to have unique IDs. For example, "Capture" cannot continue to be used as the ID for two or more different, unrelated actions in the UI. Additionally, the menus need to be responsive to display all the text that is entered as the language exception.
> 
> Can you please advise on what type of ticket needs to be submitted to ensure that Language Exceptions can be used, if that will be the required method?

**Comment by Rainribbon** [comment]
As to be expected, there has been some unhelpful back and forth with Kaltura Support regarding this issue, and they won't recognize it as an issue.

Kaltura Support: 

> I wanted to share an update from our Product team. At this time, there are no plans to support this functionality within theming. Based on their guidance, they recommend using language exceptions as a workaround in this case.

My response:

> Thanks for following up! I have already asked for a feature request ticket to revert some problematic text changes in theming.
> 
> That said, if I understand correctly, Language Exceptions are supposed to be Kaltura's supported method of updating UI text. Therefore, it should be considered a bug (not a feature request) for message IDs to be reused for unrelated UI elements. In its current state, the Language Exceptions feature is broken.

Kaltura Support:

> It is not a bug because the language exception works if you apply it. However, please note that the expected behavior of language exceptions is to change all the text elements containing the same msgid. But if you use the language exception for your case, it works. 

My response:

> It is a design flaw negatively impacting the language exceptions to have a msgid reused for unrelated functions, such that the language exceptions cannot be used to target the elements in question.
